binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtilsTests.java

```diff
@@ -16,77 +16,349 @@
-	@Test
-	void findMatchingSerdesForSimpleGenericType() {
-
-		ResolvableType geDate = ResolvableType.forType(new ParameterizedTypeReference<GenericEvent<Date>>() { });
-		ResolvableType geBounded = ResolvableType.forType(new ParameterizedTypeReference<GenericEvent<? extends Date>>() { });
-		ResolvableType geWildcard = ResolvableType.forType(new ParameterizedTypeReference<GenericEvent<?>>() { });
-		ResolvableType geRaw = ResolvableType.forRawClass(GenericEvent.class);
-
-		ApplicationContextRunner contextRunner = new ApplicationContextRunner()
-			.withUserConfiguration(SerdeResolverTestApp.class);
-
-		contextRunner.run((context) -> {
-
-			assertThat(SerdeResolverUtils.findMatchingSerdes(context, geDate))
-				.extracting("name")
-				.containsExactly("genericEventDateSerde", "genericEventDateBoundedSerde", "genericEventWildcardSerde", "genericEventRawSerde");
-
-			assertThat(SerdeResolverUtils.findMatchingSerdes(context, geBounded))
-				.extracting("name")
-				.containsExactly("genericEventDateBoundedSerde", "genericEventWildcardSerde");
-
-			assertThat(SerdeResolverUtils.findMatchingSerdes(context, geWildcard))
-				.extracting("name")
-				.containsExactly("genericEventWildcardSerde");
-
-			// Because GenericEvent is a parameterized type, Serde<GenericEvent> resolves to Serde<GenericEvent<?>>
-			// which is not assignable from GenericEvent
-			assertThat(SerdeResolverUtils.findMatchingSerdes(context, geRaw))
-				.extracting("name")
-				.isEmpty();
-		});
+	@Nested
+	class ResolveForType {
+
+		private ApplicationContextRunner contextRunner = new ApplicationContextRunner();
+
+		private Serde<?> fallback = mock(Serde.class);
+
+		@Test
+		void returnsFallbackSerdeForWildcard() {
+			this.contextRunner
+				.withConfiguration(AutoConfigurations.of(SerdeResolverSimpleTestApp.class))
+				.run((context) -> {
+					ResolvableType wildcardType = ResolvableType.forClass(Serde.class).getGeneric(0);
+					assertThat(SerdeResolverUtils.resolveForType(context, wildcardType, fallback)).isSameAs(fallback);
+				});
+		}
+
+		@Test
+		void returnsSerdeBeanForMatchingType() {
+			this.contextRunner
+				.withConfiguration(AutoConfigurations.of(SerdeResolverSimpleTestApp.class))
+				.run((context) -> {
+					ResolvableType fooType = ResolvableType.forClass(Foo.class);
+					assertThat(SerdeResolverUtils.resolveForType(context, fooType, fallback)).isInstanceOf(FooSerde.class);
+				});
+		}
+
+		@Nested
+		class NoMatchingSerdeBeans {
+
+			@ParameterizedTest
+			@MethodSource("kafkaStreamsBuiltInTypes")
+			void returnsStandardSerdeForKafkaStreamsBuiltInType(Class<?> builtInType, Serde<?> expectedBuiltInSerde) {
+				contextRunner.run((context) ->
+					assertThat(SerdeResolverUtils.resolveForType(context, ResolvableType.forClass(builtInType), fallback))
+						.isInstanceOf(expectedBuiltInSerde.getClass()));
+			}
+
+			static Stream<Arguments> kafkaStreamsBuiltInTypes() {
+				return Stream.of(
+					arguments(String.class, Serdes.String()),
```

Comment:
I did re-order these and make them match the new standard default classes in SRU